netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -185,6 +187,7 @@ public NettyChannelBuilder sslContext(SslContext sslContext) {
       GrpcSslContexts.ensureAlpnAndH2Enabled(sslContext.applicationProtocolNegotiator());
     }
     this.sslContext = sslContext;
+    this.negotiationType = NegotiationType.TLS;
```

Comment:
While I see the value in this, doing things implicitly can lead to trouble. We mostly try to have the builder be "dumb setters." Since I'm conflicted on this, let's just leave it out for now.